src/test/mediapipeflow_test.cpp

```diff
@@ -1944,6 +1948,112 @@ TEST_F(MediapipeConfigChanges, ConfigWithEmptyBasePath) {
     EXPECT_EQ(definition->getInputsInfo().count("in2"), 0);
     checkStatus<KFSRequest, KFSResponse>(modelManager, StatusCode::OK);
 }
+class MediapipeSerialization : public ::testing::Test {
+    class MockedMediapipeGraphExecutor : public ovms::MediapipeGraphExecutor {
+    public:
+        Status partialSerialize(const std::string& name, ::inference::ModelInferResponse& response, const ::mediapipe::Packet& packet) const {
+            return ovms::MediapipeGraphExecutor::partialSerialize(name, response, packet);
+        }
+
+        MockedMediapipeGraphExecutor(const std::string& name, const std::string& version, const ::mediapipe::CalculatorGraphConfig& config,
+            stream_types_mapping_t inputTypes,
+            stream_types_mapping_t outputTypes,
+            std::vector<std::string> inputNames, std::vector<std::string> outputNames,
+            const std::unordered_map<std::string, std::shared_ptr<PythonNodeResource>>& pythonNodeResources) :
+            MediapipeGraphExecutor(name, version, config, inputTypes, outputTypes, inputNames, outputNames, pythonNodeResources) {}
+    };
+
+protected:
+    MockedMediapipeGraphExecutor* executor;
+    ::inference::ModelInferResponse mp_response;
+    void SetUp() {
+        ovms::stream_types_mapping_t mapping;
+        mapping["kfs_response"] = mediapipe_packet_type_enum::KFS_RESPONSE;
+        mapping["tf_response"] = mediapipe_packet_type_enum::TFTENSOR;
+        mapping["ov_response"] = mediapipe_packet_type_enum::OVTENSOR;
+        mapping["mp_response"] = mediapipe_packet_type_enum::MPTENSOR;
+        mapping["mp_img_response"] = mediapipe_packet_type_enum::MEDIAPIPE_IMAGE;
+        const std::vector<std::string> inputNames;
+        const std::vector<std::string> outputNames;
+        const ::mediapipe::CalculatorGraphConfig config;
+        std::unordered_map<std::string, std::shared_ptr<PythonNodeResource>> pythonNodeResources;
+        executor = new MockedMediapipeGraphExecutor("", "", config, mapping, mapping, inputNames, outputNames, pythonNodeResources);
+    }
+    void TearDown() {
+        delete executor;
+    }
+};
+
+TEST_F(MediapipeSerialization, KFSResponse) {
+    KFSResponse response;
+    response.set_id("1");
+    auto output = response.add_outputs();
+    output->add_shape(1);
+    output->set_datatype("FP32");
+    std::vector<float> data = {1.0f};
+    response.add_raw_output_contents()->assign(reinterpret_cast<char*>(data.data()), data.size() * sizeof(float));
+    ::mediapipe::Packet packet = ::mediapipe::MakePacket<KFSResponse*>(&response);
+    ASSERT_EQ(executor->partialSerialize("kfs_response", mp_response, packet), StatusCode::OK);
+    ASSERT_EQ(mp_response.id(), "1");
+    auto mp_output = mp_response.outputs(0);
```

Comment:
First check if there is exactly 1 output. The same goes for line 2001 - check if there is 1 raw output content

Reply:
The same goes for all other tests before accessing nth element